sources/packages/api/src/services/institution-location/institution-location.service.ts

```diff
@@ -116,19 +116,9 @@ export class InstitutionLocationService extends RecordDataModelService<Instituti
       .getOne();
   }
 
-  //Query to be replaced, when the authorization is in place
-  async getAllUserLocations(userName: string): Promise<UserLocationDto[]> {
-    return this.repo
-      .query(`SELECT institutionlocations.id, institutionlocations.NAME
-              FROM            sims.users USERS 
-              LEFT JOIN       sims.institution_users INSTITUTIONUSER 
-              ON              institutionuser.user_id = users.id 
-              LEFT JOIN       sims.institution_user_auth INSTITUTIONUSERAUTH 
-              ON              institutionuserauth.institution_user_id = institutionuser.id 
-              LEFT OUTER JOIN sims.institution_locations INSTITUTIONLOCATIONS 
-              ON              institutionlocations.institution_id = institutionuser.institution_id 
-              AND             (institutionlocations.id = institutionuserauth.institution_location_id 
-              OR              institutionuserauth.institution_location_id IS NULL )
-              WHERE           users.user_name = '${userName}'`);
+  async getMyInstitutionlocations(
+    locationIds: number[],
+  ): Promise<InstitutionLocationsDetailsDto[]> {
```

Comment:
I believe that this method should return the entity returned from the DB.
If we are going to consume this method anywhere else it would be good to not be restricted to what the API is returning to this specific method through the DTO.
IMHO, the service layer should be more generic.